aioesphomeapi/_frame_helper/noise.py

```diff
@@ -357,14 +358,23 @@ def _handle_frame(self, frame: bytes) -> None:
                 )
             )
             return
+        msg_length = len(msg)
+        msg_cstr = msg
+        if msg_length < 4:
+            self._handle_error_and_close(
+                ProtocolAPIError(
+                    f"{self._log_name}: Decrypted message too short: {msg_length} bytes"
+                )
+            )
+            return
```

Comment:
We don't actually have coverage for this case but it could happen in theory.... although its very unlikely